### PR TITLE
 New tests, bug fixes, random++

### LINKED ISSUE #27: ig "[noun] [randomdate]" fails with: Cannot validate argument on parameter 'Name'. The argument is null or empty
## Steps

* NameIT module version 2.1.0
* Run `NameIT\Invoke-Generate "[noun] [randomdate]" -Count 1`

## Result

```
Test-GeneratorInSet : Cannot validate argument on parameter 'Name'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
At C:\Users\jpoehls\Documents\PowerShell\Modules\nameit\2.1.0\Public\Invoke-Generate.ps1:126 char:47
+                 if (Test-GeneratorInSet -Name $fn) {
+                                               ~~~
+ CategoryInfo          : InvalidData: (:) [Test-GeneratorInSet], ParameterBindingValidationException
+ FullyQualifiedErrorId : ParameterArgumentValidationError,Test-GeneratorInSet
```

## $PSVersionTable

```
Name                           Value
----                           -----
PSVersion                      6.2.0
PSEdition                      Core
GitCommitId                    6.2.0
OS                             Microsoft Windows 10.0.15063
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

## Notes

Possibly related to the custom date format I have set in Windows? `yyyy-MM-dd`

For example, `ig "[randomdate]` returns `0183-11-22`.

### LINKED ISSUE #28: Arguments with spaces, even quoted, will fail
Demonstration:

This works:
```powershell
ig "[randomdate 1/1/1999 12/21/1999 'yyyy-MM']" -Count 1
```

This does not:
```powershell
ig "[randomdate 1/1/1999 12/21/1999 'yyyy MM']" -Count 1
```

The newer parsing code was written with this in mind, expecting to handle quoted content and pass it along appropriately, but this was stymied by not quoting the literal quotes 🤦‍♂ 

At least that makes the fix (forthcoming) easy.

### PR DESCRIPTION
**Pull request by briantist**
- Add tests for reported and discovered bugs
- Add `-SetSeed` parameter to `Invoke-Generate` to aid in repeatable runs
(for tests mostly)
- Align `randomdate` generator to use `Get-Random` so that it inherits a set
seed
- Fix #27 (literal spaces aren't processed properly)
- Fix #28 (arguments with spaces fail even when specified properly)
- Verstion 2.1.1

### CODE REVIEW

File: Private/GeneratorSet/Test-GeneratorInSet.ps1 [hunk 1]
```diff
@@ -6,7 +6,8 @@ function Test-GeneratorInSet {
             Mandatory ,
             ValueFromPipeline
         )]
-        [ValidateNotNullOrEmpty()]
+        [ValidateNotNull()]
+        [AllowEmptyString()]
```
Comment (briantist) [inline Private/GeneratorSet/Test-GeneratorInSet.ps1]: These changes are what fixes #27

File: Public/Invoke-Generate.ps1 [hunk 2]
```diff
@@ -129,7 +137,7 @@ function Invoke-Generate {
                         Select-Object -Skip 1 |
                         ForEach-Object -Process {
                             if ($_.Type -eq [System.Management.Automation.PSTokenType]::String) {
-                                '{0}' -f [System.Management.Automation.Language.CodeGeneration]::EscapeSingleQuotedStringContent($_.Content)
+                                "'{0}'" -f [System.Management.Automation.Language.CodeGeneration]::EscapeSingleQuotedStringContent($_.Content)
```
Comment (briantist) [inline Public/Invoke-Generate.ps1]: This fixes #28

File: __tests__/nameit.tests.ps1 [hunk 3]
```diff
@@ -2,6 +2,7 @@ $manifestPath = $PSScriptRoot | Join-Path -ChildPath '..' | Join-Path -ChildPath
 Import-Module $manifestPath -Force
 
 Describe "NameIT Tests" {
+    $StaticSeed = [System.Random]::new().Next()
```
Comment (briantist) [inline __tests__/nameit.tests.ps1]: Originally chose a constant magic seed, but this kind of screws up `Get-Random` for everything that runs in the same session after the tests run. Fine for AppVeyor, not so great for your dev machine. Decided instead to randomly produce a new seed at the beginning of the tests to use throughout testing. Should retain decent pseudorandomness post tests.

**Comment by dfinke** [comment]
Well, the challenge is, this grew organically, no tests from the outset. This will probably become a Whac-A-Mole problem.

I don't know if these bugs existed before the new changes?

**Comment by briantist** [comment]
@dfinke yeah I think as we add new functionality, some existing bugs that weren't likely to be exposed are seeing the light. Of he 2 bugs fixed in this PR only #27 is a regression. 

Also hoping as we add new tests and get better at adding tests beforehand, we'll be able to prevent regressions and new bugs before they land. 

The `-SetSeed` parameter here helps in writing some tests by being able to get consistent output that can be compared for equality (you can see it used in some of the new tests), so this should lead to a boon of new tests that would have been difficult before. 🎉 to learning from mistakes!

**Pull request merged**